sklearn/utils/_metadata_requests.py

```diff
@@ -403,7 +403,7 @@ def _check_warnings(self, *, params):
                 "warning, or to True to consume and use the metadata."
             )
 
-    def _route_params(self, params):
+    def _route_params(self, params, parent, caller_method):
```

Comment:
the `parent, caller` order is not consistent here. I think `parent, caller` makes more sense than `caller, parent`. Also, `caller` would be enought, doesn't need to be `caller_method`.